modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,758 @@
+            ? this.rtcMuteTimeout
+            : this.conference.isP2PActive() ? this.p2pRtcMuteTimeout : this.outOfLastNTimeout;
+    }
+
+    /**
+     * Initializes <tt>TrackStreamingStatus</tt> and bind required event
+     * listeners.
+     */
+    init() {
+        // Handles P2P status changes
+        this._onP2PStatus = this.figureOutConnectionStatus.bind(this);
+        this.conference.on(JitsiConferenceEvents.P2P_STATUS, this._onP2PStatus);
+
+        // Used to send analytics events for the participant that left the call.
+        this._onUserLeft = this.onUserLeft.bind(this);
+        this.conference.on(JitsiConferenceEvents.USER_LEFT, this._onUserLeft);
+
+        // On some browsers MediaStreamTrack trigger "onmute"/"onunmute"
+        // events for video type tracks when they stop receiving data which is
+        // often a sign that remote user is having connectivity issues
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+
+            this._onTrackRtcMuted = this.onTrackRtcMuted.bind(this);
+            this.rtc.addListener(
+                RTCEvents.REMOTE_TRACK_MUTE, this._onTrackRtcMuted);
+
+            this._onTrackRtcUnmuted = this.onTrackRtcUnmuted.bind(this);
+            this.rtc.addListener(
+                RTCEvents.REMOTE_TRACK_UNMUTE, this._onTrackRtcUnmuted);
+
+            // Listened which will be bound to JitsiRemoteTrack to listen for
+            // signalling mute/unmute events.
+            this._onSignallingMuteChanged
+                = this.onSignallingMuteChanged.bind(this);
+            this.track.on(
+                JitsiTrackEvents.TRACK_MUTE_CHANGED,
+                this._onSignallingMuteChanged);
+
+            // Used to send an analytics event when the video type changes.
+            this._onTrackVideoTypeChanged
+                = this.onTrackVideoTypeChanged.bind(this);
+            this.track.on(
+                JitsiTrackEvents.TRACK_VIDEOTYPE_CHANGED,
+                videoType => this._onTrackVideoTypeChanged(videoType));
+        }
+
+        this._onForwardedSourcesChanged = this._onForwardedSourcesChanged.bind(this);
+        this.conference.on(
+            JitsiConferenceEvents.FORWARDED_SOURCES_CHANGED,
+            this._onForwardedSourcesChanged);
+
+        this._onLastNValueChanged
+            = this.figureOutConnectionStatus.bind(this);
+        this.rtc.on(
+            RTCEvents.LASTN_VALUE_CHANGED, this._onLastNValueChanged);
+    }
+
+    /**
+     * Removes all event listeners and disposes of all resources held by this
+     * instance.
+     */
+    dispose() {
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+            this.rtc.removeListener(
+                RTCEvents.REMOTE_TRACK_MUTE,
+                this._onTrackRtcMuted);
+            this.rtc.removeListener(
+                RTCEvents.REMOTE_TRACK_UNMUTE,
+                this._onTrackRtcUnmuted);
+
+            this.track.off(
```

Comment:
The logic to handle the `JitsiConferenceEvents.TRACK_REMOVED` event has been moved into the `dispose` method of this class.

Reply:
And when a track is removed this instance will be disposed anyway, right?

Reply:
I will double check to confirm if that is true. If not, I can dispose it where we are emitting `JitsiConferenceEvents.TRACK_REMOVED`

Reply:
I don't think I saw any logic in LJM that called `dispose` when a tracked was removed. I wasn't sure if this is intended so I added logic to clear `_trackStreamingStatusImpl` and `_trackStreamingStatus` where the event was emitted